experimental/sql/src/alg_to_ssa.py

```diff
@@ -133,6 +133,16 @@ def match_and_rewrite(self, op: RelAlg.BinOp, rewriter: PatternRewriter):
 @dataclass
 class ProjectRewriter(RelAlgRewriter):
 
+  def find_type_of_expression(self, op: RelAlg.Expression,
+                              input_bag: RelSSA.Bag) -> RelSSA.DataType:
+    if isinstance(op, RelAlg.Column):
+      return input_bag.lookup_type_in_schema(op.col_name.data)
+    if isinstance(op, RelAlg.Literal):
+      return op.type
+    if isinstance(op, RelAlg.BinOp):
+      return self.find_type_of_expression(op.lhs.op, input_bag)
```

Comment:
I suppose that this works because all `BinOp`s only accept two operands of the same type? Add a comment about this assumption?